resources/dw/dataviz/schema_dataviz.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Cloudera. All Rights Reserved.
+//
+// This file is licensed under the Apache License Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package dataviz
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+)
+
+var datavizSchema = schema.Schema{
+	MarkdownDescription: "Cloudera Data Warehouse (CDW) integrates [Data Visualization](https://docs.cloudera.com/data-warehouse/cloud/managing-warehouses/topics/dw-use-data-visualization.html) for building graphic representations of data, dashboards, and visual applications based on CDW data.",
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed: true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"cluster_id": schema.StringAttribute{
+			Required:            true,
+			MarkdownDescription: "The id of the CDW Cluster which the Data Visualization is attached to.",
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"name": schema.StringAttribute{
+			Required:            true,
+			MarkdownDescription: "The name of the Data Visualization.",
+		},
+
+		"image_version": schema.StringAttribute{
+			Optional:            true,
+			Computed:            true,
+			MarkdownDescription: "The version of the Data Visualization.",
+		},
+		"resource_template": schema.StringAttribute{
+			Optional: true,
+			Computed: true,
+			// TODO vcsomor add validation logic to the allowed types
+			MarkdownDescription: "The name of the resource template being used. Available options: reduced, medium, large. Empty means the default resources template will be assigned.",
+		},
+
+		"user_groups": schema.StringAttribute{
+			Optional:            true,
+			MarkdownDescription: "List of the LDAP groups which have access to this Data Visualization instance.",
+		},
+		"admin_groups": schema.StringAttribute{
```

Comment:
Can you check to see if this is a valid case for ListAttribute or StringAttribute
I found this https://developer.hashicorp.com/terraform/plugin/framework/handling-data/attributes/list